payload/builder/payload.go

```diff
@@ -215,5 +215,22 @@ func (pb *PayloadBuilder) getPayload(
 	if envelope.GetExecutionPayload().Withdrawals == nil {
 		return nil, ErrNilWithdrawals
 	}
+
+	// // TODO(pectra): Unmarshal the envelope.requests into WithdrawalRequests/DepositReceipts
```

Comment:
tbf I think this could be moved to somehwere within the engine client, as it's the engine's "job/responsibility" to return valid data